working_with_the_llvm_ast.md

```diff
@@ -0,0 +1,215 @@
+## Working With the LLVM AST and Using it to Make Changes to Clang-Tidy Source
+
+There are situations where a particular clang-tidy check will report a lot of false positives or it will require some
+additions to add more features to it based on the usecase. Solving either problem will require making changes to
+clang-tidy source and require both understanding the AST and working with the AST to make code changes as all the
+matchers in clang-tidy source are reliant on the AST and written using the AST. The following sections go into resources
+and examples that will be helpful in gaining a better understanding of the LLVM AST and writing code using it. 
+
+### Understanding the AST
+
+The following list of resources goes into the basics and details of understanding the LLVM AST - 
+
+1. LLVM AST basics, traversing the AST, important terms and examples. The link also contains a video on the AST and is
+   useful to watch - [Understanding the Clang AST](https://jonasdevlieghere.com/understanding-the-clang-ast/).
```

Comment:
In all these resources, there's a duplication of the beginning and the end. Can the you combine the hyperlink with the beginning to make this cleaner?

```
[Understanding the Clang AST](https://jonasdevlieghere.com/understanding-the-clang-ast/) - Goes over the LLVM AST basics, tra...
```